pages/api/trips/[id].js

```diff
@@ -12,8 +12,11 @@ export default async function handler(request, response) {
       return response.status(404).json({ status: "Not found" });
     }
     response.status(200).json(trip);
-  }
-  if (request.method === "DELETE") {
+  } else if (request.method === "PUT") {
+    const updatedTrip = request.body;
+    await Trip.findByIdAndUpdate(id, updatedTrip);
+    response.status(200).json({ status: `Trip ${id} successfully updated!` });
+  } else if (request.method === "DELETE") {
```

Comment:
To make your code more stable, you can think about a consistent error handling 👮‍♀️
_Here is an example which you could adapt for the other methods._

```js
else if (request.method === "PUT") {
  try {
    const updatedTrip = request.body;
    await Trip.findByIdAndUpdate(id, updatedTrip);
    return response.status(200).json({ status: `Trip ${id} successfully updated!` });
  } catch (error) {
    console.error("PUT /api/trips/:id", error);
    return response.status(500).json({ message: "Error updating trip" });
  }
```